ext/har/types.go

```diff
@@ -0,0 +1,414 @@
+        Headers:     parseStringArrMap(req.Header),
+        QueryString: parseStringArrMap(req.URL.Query()),
+        BodySize:    req.ContentLength,
+        HeadersSize: -1,
+    }
+
+    if captureContent && (req.Method == "POST" || req.Method == "PUT") {
+        log.Printf("ParseRequest: creating PostData, hasBody=%v, hasGetBody=%v", 
+            req.Body != nil, req.GetBody != nil)
+
+        harRequest.PostData = &PostData{
+            MimeType: req.Header.Get("Content-Type"),
+        }
+        
+        var body []byte
+        var err error
+
+        if req.GetBody != nil {
+            log.Printf("ParseRequest: using GetBody")
+            bodyReader, err := req.GetBody()
+            if err == nil {
+                body, err = io.ReadAll(bodyReader)
+                if err != nil {
+                    log.Printf("ParseRequest: error reading from GetBody: %v", err)
+                } else {
+                    harRequest.PostData.Text = string(body)
+                    log.Printf("ParseRequest: successfully read from GetBody: %s", string(body))
+                }
+                bodyReader.Close()
+            } else {
+                log.Printf("ParseRequest: error getting fresh body: %v", err)
+            }
+        } else if req.Body != nil {
+            log.Printf("ParseRequest: reading from Body")
+            body, err = io.ReadAll(req.Body)
+            if err != nil {
+                log.Printf("ParseRequest: error reading Body: %v", err)
+            } else {
+                // Restore the body
+                req.Body = io.NopCloser(bytes.NewBuffer(body))
+                harRequest.PostData.Text = string(body)
+                log.Printf("ParseRequest: successfully read body: %s", string(body))
+            }
+        }
+    }
+
+    return &harRequest
+}
+
+
+func (entry *Entry) fillIPAddress(req *http.Request) {
+    host := req.URL.Hostname()
+    
+    // try to parse the host as an IP address
+    if ip := net.ParseIP(host); ip != nil {
+        entry.ServerIpAddress = ip.String()
+        return
+    }
+    
+    // If it's not an IP address, perform a DNS lookup with a timeout
+    resolver := &net.Resolver{}
+    ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+    defer cancel()
+    
+    ips, err := resolver.LookupIP(ctx, "ip", host)
+    if err != nil {
+        // If lookup fails, just use the hostname
+        entry.ServerIpAddress = host
+        return
+    }
+    
+    // Prefer IPv4, but fall back to IPv6 if necessary
+    for _, ip := range ips {
+        if ipv4 := ip.To4(); ipv4 != nil {
+            entry.ServerIpAddress = ipv4.String()
+            return
+        }
+    }
+    
+    // If no IPv4 address found, use the first IP (IPv6) in the list
+    if len(ips) > 0 {
+        entry.ServerIpAddress = ips[0].String()
+    } else {
+        // If no IPs found, fall back to the hostname
+        entry.ServerIpAddress = host
+    }
```

Comment:
Server IP address is an optional field, leave it empty if net.ParseIP() fails, as I explained in the previous comment (https://github.com/elazarl/goproxy/pull/610#discussion_r1900122992).
Don't call net.Resolver at all.